### PR TITLE
Upgraded ingress-nginx stack to use Helm chart version 4.0.13

### PR DESCRIPTION
**Pull request by v-ctiutiu**
## BACKGROUND

Nginx Ingress Controller application version is a little bit too old. The current version used by the marketplace stack is `1.0.4`. Latest one at this time of writing is `1.1.0`. Other than that, the `ingress-nginx` stack should be aligned with the others, and use the Helm based deployment/upgrade/uninstall scripts.

For the Helm based installation, the `ingress-nginx` chart version `4.0.13` corresponds to Nginx application version `1.1.0`.

-----------------------------------------------------------------------

## Changes

1. Used the stack generator script provided in the main repository (`./stack-templates/generate-stack.sh`) to create the required scripts (`deploy.sh`, `upgrade.sh`, `uninstall.sh`), as well as the Helm values file.
2. Validated that the stack scripts work as expected for each scenario (deploy/upgrade/uninstall).
3. Created main `README`, containing a quick start guide for the Nginx Ingress stack. Also, make sure to point the user to additional resources and guides, to learn more about what Nginx can do.

-----------------------------------------------------------------------

## Checklist
- [x] review the [contributing doc](https://github.com/digitalocean/marketplace-kubernetes/blob/master/CONTRIBUTING.md) with steps for both adding or updating your application (if applicable)
------------------------------------------------------------------------

Reviewer: @marketplace-eng

### CODE REVIEW

File: stacks/ingress-nginx/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+- Log traffic from both controllers during this changeover.
+- Uninstall the old controller once traffic has fully drained from it.
+- For details on all of these steps see [Upgrading With Zero Downtime in Production](#upgrading-with-zero-downtime-in-production).
+
+### Uninstalling the Nginx Ingress Controller
+
+To delete your installation of NGINX Ingress Controller, please run the following `Helm` command:
+
+```console
+helm uninstall ingress-nginx -n ingress-nginx
```
Comment (caquillo07) [inline stacks/ingress-nginx/README.md:150]: should we also add `kubectl delete ns ingress-nginx` as part of the instructions?

**Review by caquillo07 — COMMENTED** [review]
Looks good, only thing I noticed is the upgrade and uinstall scripts do not have executing permissions 
```shell
➜  marketplace-kubernetes-v-ctiutiu-issue319 git:(master) ✗ ll ./stacks/ingress-nginx/
drwxrwxr-x hector staff 224 B  Tue Feb 15 12:05:04 2022  .
drwxrwxr-x hector staff 1.5 KB Tue Feb 15 12:05:04 2022  ..
.rwxr-xr-x hector staff 1.2 KB Tue Feb 15 12:05:04 2022  deploy.sh
.rw-rw-r-- hector staff  10 KB Tue Feb 15 12:05:04 2022  README.md
.rw-rw-r-- hector staff 292 B  Tue Feb 15 12:05:04 2022  uninstall.sh
.rw-rw-r-- hector staff 983 B  Tue Feb 15 12:05:04 2022  upgrade.sh
.rw-rw-r-- hector staff 620 B  Tue Feb 15 12:05:04 2022  values.yml
```
If this is expected, disregard. I do have a suggestion/question for the uinstall instructions though

File: stacks/ingress-nginx/README.md (hunk 1 above)
Comment (v-ctiutiu) [reply stacks/ingress-nginx/README.md:150]: Good point. Should be fixed in the latest commit (added a note in the README).

**Review by caquillo07 — APPROVED** [review]
Looks great, thanks for making the changes!

**Pull request merged**